### PR TITLE
Automate with Gulp

### PR DESCRIPTION
**Pull request by mdo**
Adds a basic Gulpfile to handle minification. Doing so involved moving the `github-btn.source.html` button to a `src/` directory and renaming it `github-btn.html`. Anything else would require more Gulp dependencies to move things around and rename them.

Unsure what else needs going just yet, but I'll start a list:

- [x] Update readme to include instructions
- [x] Update to use original HTML minifier
- [x] Split source file into HTML, CSS, JS and use Gulp to compile

/cc @XhmikosR

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+var gulp    = require('gulp');
+var htmlmin = require('gulp-html-minifier');
+
+// Minify HTML
+gulp.task('minify', function() {
+  gulp.src('./src/github-btn.html')
+    .pipe(htmlmin({
+      collapseWhitespace: true,
+      minifyCSS: true,
+      minifyJS: true
```
Comment (XhmikosR) [inline gulpfile.js]: You need to add to remove optional tags in order to match the previous behavior.

**Comment by XhmikosR** [comment]
Definitely a good step forward :+1: 

But, IMO splitting stuff will allow us to run lint for JS, html validation, etc.

File: package.json [hunk 2]
```diff
@@ -21,5 +21,9 @@
   "files": [
     "github-btn.html",
     "LICENSE.md"
-  ]
+  ],
+  "devDependencies": {
+    "gulp": "^3.8.11",
+    "gulp-html-minifier": "^0.1.6",
```
Comment (XhmikosR) [inline package.json]: This comma breaks stuff since it's not valid :P

Don't forget to squash when you are done.

Comment (XhmikosR) [inline package.json]: Oh, also, you seem to use a fork of the original repo... https://github.com/jonschlinkert/gulp-htmlmin

**Comment by XhmikosR** [comment]
On a side note, I'd say just use `gulp-rename` and keep the src file in root for now.

File: package.json (hunk 2 above)
Comment (XhmikosR) [inline package.json]: So basically,

```diff
 gulpfile.js  | 2 +-
 package.json | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/gulpfile.js b/gulpfile.js
index c60b8d7..707ed64 100644
--- a/gulpfile.js
+++ b/gulpfile.js
@@ -1,5 +1,5 @@
 var gulp    = require('gulp');
-var htmlmin = require('gulp-html-minifier');
+var htmlmin = require('gulp-htmlmin');
 
 // Minify HTML
 gulp.task('minify', function() {
diff --git a/package.json b/package.json
index f8a489e..48e0805 100644
--- a/package.json
+++ b/package.json
@@ -24,6 +24,6 @@
   ],
   "devDependencies": {
     "gulp": "^3.8.11",
-    "gulp-html-minifier": "^0.1.6",
+    "gulp-htmlmin": "^1.1.1"
   }
 }

```

**Comment by XhmikosR** [comment]
Oh, also, if you go this way with the src dir, don't forget to add it in _config.yml excludes along with `node_modules`.

**Comment by XhmikosR** [comment]
Here's my gulp branch with everything rebased and working. The compiled html is identical: https://github.com/mdo/github-buttons/compare/master...XhmikosR:gulp

Just add my fork as a remote, git reset your gulp branch to mine, and then force push.

**Comment by mdo** [comment]
Closing for #87. Woot!

**Pull request closed without merging**